Content.Server/Body/Systems/RespiratorSystem.cs

```diff
@@ -296,11 +387,37 @@ bool CanMetabolize(EntityEffect effect)
         return saturation;
     }
 
+    /// <summary>
+    /// Returns a saturation value for a given air mixture for a specific respirator.
+    /// </summary>
+    private float GetSaturation(Entity<RespiratorComponent> entity, GasMixture gas)
+    {
+        var moles = gas[(int)entity.Comp.MetabolizedGas];
+
+        return Math.Max(0, moles * entity.Comp.SaturationMultiplier);
+    }
+
+    /// <summary>
+    /// Returns a saturation value for the air mixture inside a respirator's lungs
+    /// </summary>
+    private float GetSaturation(Entity<RespiratorComponent> entity)
+    {
+        return GetSaturation(entity, entity.Comp.Air);
+    }
+
     private void TakeSuffocationDamage(Entity<RespiratorComponent> ent)
     {
         if (ent.Comp.SuffocationCycles == 2)
             _adminLogger.Add(LogType.Asphyxiation, $"{ToPrettyString(ent):entity} started suffocating");
 
+        _damageableSys.TryChangeDamage(ent, ent.Comp.Damage, interruptsDoAfters: false);
+
+        if (!HasComp<BodyComponent>(ent))
+        {
+            _alertsSystem.ShowAlert(ent, ent.Comp.Alert);
+            return;
+        }
```

Comment:
this feels like it should be based on an event being handled or not, but also might be included in my concerns about incomplete fallback